### ISSUE TITLE
Updating `git` version used in Docker build strategy

### ISSUE THREAD

**Issue by alikhajeh1** [openshift/origin#15871]
openshift/origin-pod:v1.5.1 is currently using git version 1.8.3.1. Are there any plans to update it, e.g. to patch the security vulnerabilities mentioned in https://lkml.org/lkml/2017/8/10/757, in build configs that use the Docker build strategy? If not, is there a recommended way for us to update git ourself (e.g. create a custom image based on openshift/origin-docker-builder?)

##### Version
```
oc v1.5.1
kubernetes v1.5.2
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server https://dashboard.abarcloud.internal:8443
openshift v1.5.1
kubernetes v1.5.2
```

**Comment by mfojtik** [comment]
@alikhajeh1 i guess the docker builder and s2i builders use the latest GIT version available in centos7 repos (mine shows 1.8.3.1-6.el7_2.1)

**Comment by bparees** [comment]
that's correct, it's from centos repos.  we don't backport to origin releases generally speaking, so to get a new one you'd have to go to a new release that uses a newer image that picked up a newer version, if there is one available.

As for updating it yourself, yes you can absolutely do a FROM to extend the existing docker-builder and sti-builder images, but you'll need to make sure that the new image has the right tag name and exists locally on all your cluster nodes to ensure no cluster node is forced to pull the image manually (at which point it would pull from dockerhub and not get your customized version)
